File: src/debug/entry/javascript/problems/116.js
function runUserScript(userFunm, params, paramTypes) {
    if (params.length !== paramTypes.length) {
        onParameterError();
    }

    const parsedParams = params.map((param, index) => {
        const type = paramTypes[index];
        return parseParameter(index, type, param);
    });
    userFunm.apply(null, parsedParams);
}

function Node(val, left, right, next) {
    this.val = val;
    this.left = left;
    this.right = right;
    this.next = next;
}

/**
 * @param {number[]} param
 */
function parseNode(nums) {
    const arr = [];
    nums.map((n, i) => {
        const node = new Node(n);
        arr.push(node);
        if (i !== 0) {
            if (i % 2 === 1) {
                arr[(i - 1) / 2].left = node;
            } else {
                arr[(i - 2) / 2].right = node;
            }
        }
    });
    return arr[0];
}

function parseSpecialParameter(index, type, param) {
    switch (type) {
        case "Node":
            return parseNode(param);
    }
    return null;
}

File: src/debug/entry/python3/problems/116.py
def runUserScript(func, params, paramTypes):
    if (len(params) != len(paramTypes)):
        onParameterError()

    newParams = []
    for i, val in enumerate(params):
        newParams.append(parseParameter(i, paramTypes[i], val))
    func(*newParams)


class Node:
    def __init__(self, val, left, right, next):
        self.val = val
        self.left = left
        self.right = right
        self.next = next


def parseNode(param):
    arr = []

    for i, val in enumerate(param):
        node = Node(val, None, None, None)
        arr.append(node)
        if i is not 0:
            if i % 2 is 1:
                arr[int((i - 1) / 2)].left = node
            else:
                arr[int((i - 2) / 2)].right = node

    return arr[0]


def parseSpecialParameter(index, paramType, param):
    if paramType == "Node":
        return parseNode(param)
    return None
